### PR TITLE
Progress Bar

### PR DESCRIPTION
**Pull request by gcater**
https://www.loom.com/share/e48d2c3aebc9462f9e3ed53d89918634?sid=cfcc19ff-6de4-4fea-80b1-2cf490f03ced

Screenshot 2024-04-18 at 4 00 01 PM
Screenshot 2024-04-18 at 4 00 11 PM

Adding Progress Bar to the section index. When the window changes the spacing does not change the same way Network state does. When a user scrolls down a section in Network State the progress bar moves along as well. Need to implement this functionality.

### CODE REVIEW

File: src/components/ProgressBar.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import { api } from "~/utils/api";
+
+const ProgressBar = ({ bookId, chapterId,sectionId }: { bookId: string; chapterId: string; sectionId: string }) => {
+  const { data: retrievedBook } = api.book.getBook.useQuery(bookId);
+
+  if (!retrievedBook) return <p>No book found!</p>;
+
+  const currentChapterIndex = retrievedBook.chapters.findIndex(chapter => chapter.id === chapterId);
+  const currentChapterTitle = retrievedBook.chapters[currentChapterIndex]?.title ?? 'Unknown Chapter';
+  const currentChapter = retrievedBook.chapters[currentChapterIndex];
```
Comment (Evanc123) [inline src/components/ProgressBar.tsx:12]: Switch the order of this line and the line above

Comment (Evanc123) [reply src/components/ProgressBar.tsx:12]: Saves a check

**Review by Evanc123 — APPROVED** [review]
Reasonable!

File: src/components/ProgressBar.tsx (hunk 1 above)
Comment (gcater) [reply src/components/ProgressBar.tsx:12]: const currentChapterIndex = retrievedBook.chapters.findIndex((chapter) => chapter.id === chapterId,);
const currentChapter = retrievedBook.chapters[currentChapterIndex];
const currentChapterTitle = currentChapter?.title;
  
Like this?

Comment (Evanc123) [reply src/components/ProgressBar.tsx:12]: Yeah

**Pull request merged**